Add Embeddings to API Reference (#96)
kluster-ai/docs: api-reference/reference.md

---

## Embeddings

### Create embeddings

`POST https://api.kluster.ai/v1/embeddings`

Creates an embedding vector representing the input text. Machine learning models and algorithms can easily consume these vector representations to understand semantic relationships between pieces of text.

<div class="grid" markdown>
<div markdown>

**Request**

`input` ++"string or array"++ <span class="required" markdown>++"required"++</span>

Input text to embed, encoded as a string or array of tokens. To embed multiple inputs in a single request, pass an array of strings or an array of token arrays. The input must not exceed the max input tokens for the model (8192 tokens for `BAAI/bge-m3`), it cannot be an empty string, and any array must be 2048 dimensions or less.

---

`model` ++"string"++ <span class="required" markdown>++"required"++</span>

ID of the model to use.

---

`encoding_format` ++"string"++

The format to return the embeddings in. Can be either `float` or `base64`. Defaults to `float`.

---

> Please note, the embeddings endpoint doesn’t support the dimensions parameter. Models such as `BAAI/bge‑m3` always return a fixed 1024‑dimensional vector. Supplying dimensions will trigger a `400 Bad Request`.

Reviewer: ```suggestion
`dimensions` <span class="not-supported" markdown>++"not supported"++</span>

Please note, the embeddings endpoint doesn’t support the `dimensions` parameter. Models such as `BAAI/bge‑m3` always return a fixed 1024‑dimensional vector. Supplying `dimensions` will trigger a `400 Bad Request`.
```

I feel like this is getting lost a little bit. What do you think about doing something like this?

Author: Looks great!

---

## Embeddings

### Create embeddings

`POST https://api.kluster.ai/v1/embeddings`

> Creates an embedding vector representing the input text. These vector representations can be easily consumed by machine learning models and algorithms to understand semantic relationships between pieces of text.

Reviewer: ```suggestion
Creates an embedding vector representing the input text. Machine learning models and algorithms can easily consume these vector representations to understand semantic relationships between pieces of text.
```

<div class="grid" markdown>
<div markdown>

**Request**

`input` ++"string or array"++ <span class="required" markdown>++"required"++</span>

> Input text to embed, encoded as a string or array of tokens. To embed multiple inputs in a single request, pass an array of strings or array of token arrays. The input must not exceed the max input tokens for the model (8192 tokens for `BAAI/bge-m3`), cannot be an empty string, and any array must be 2048 dimensions or less.

Reviewer: ```suggestion
Input text to embed, encoded as a string or array of tokens. To embed multiple inputs in a single request, pass an array of strings or an array of token arrays. The input must not exceed the max input tokens for the model (8192 tokens for `BAAI/bge-m3`), it cannot be an empty string, and any array must be 2048 dimensions or less.
```

---

`model` ++"string"++ <span class="required" markdown>++"required"++</span>

ID of the model to use.

---

`dimensions` ++"integer"++

The number of dimensions the resulting output embeddings should have.

---

`encoding_format` ++"string"++

The format to return the embeddings in. Can be either `float` or `base64`. Defaults to `float`.

---

`user` ++"string"++

A unique identifier representing your end-user.

---

**Returns**

> A list of [embedding objects](#embedding-object).

Reviewer: ```suggestion
A list of [Embedding objects](#embedding-object).
```

Capitalize here because it should match the title of the section we're linking to

        -H "Authorization: Bearer $API_KEY" \
        -H "Content-Type: application/json" \
        -d '{
        "input": "The food was delicious and the waiter...",
        "model": "BAAI/bge-m3",
        "encoding_format": "float"
        }'
    ```

```Json title="Response"
{
  "object": "list",
  "created": 1744935248,
  "model": "BAAI/bge-m3",
  "data": [
    {
      "object": "embedding",
      "index": 0,
      "embedding": [
        0.00885772705078125,
        -0.0010204315185546875,
        -0.045135498046875,
        0.00478363037109375,
        -0.02642822265625,
        /* ... truncated for brevity ... */
        -0.0168304443359375,
        0.0229339599609375,
        0.007648468017578125,
        -0.03875732421875,
        0.05487060546875
      ]
    }
  ],
  "usage": {
    "prompt_tokens": 13
  }
}
```

</div>
</div>

---

### Embedding object

Represents an embedding vector returned by the embeddings endpoint.

<div class="grid" markdown>
<div markdown>

**Properties**

`embedding` ++"array"++

The embedding vector, which is a list of floats.

---

`index` ++"integer"++

The index of the embedding in the list of embeddings.

---

`object` ++"string"++

> The object type, which is always embedding.

Reviewer: ```suggestion
The object type, which is always `"embedding"`.
```
